config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/JmsDestination.java

```diff
@@ -0,0 +1,338 @@
+        return providerType;
+    }
+
+    public void setProviderType(String providerType) {
+        this.providerType = providerType;
+    }
+
+    public String getInitialContextFactoryClassName() {
+        return initialContextFactoryClassName;
+    }
+
+    public void setInitialContextFactoryClassName(String initialContextFactoryClassName) {
+        this.initialContextFactoryClassName = initialContextFactoryClassName;
+    }
+
+    public Map<String, Object> getJndiProperties() {
+        return jndiProperties;
+    }
+
+    public void setJndiProperties(Map<String, Object> jndiProperties) {
+        this.jndiProperties = jndiProperties;
+    }
+
+    public String getJndiUrl() {
+        return jndiUrl;
+    }
+
+    public void setJndiUrl(String jndiUrl) {
+        this.jndiUrl = jndiUrl;
+    }
+
+    public String getJndiUsername() {
+        return jndiUsername;
+    }
+
+    public void setJndiUsername(String jndiUsername) {
+        this.jndiUsername = jndiUsername;
+    }
+
+    public String getJndiPasswordRef() {
+        return jndiPasswordRef;
+    }
+
+    public void setJndiPasswordRef(String jndiPasswordRef) {
+        this.jndiPasswordRef = jndiPasswordRef;
+    }
+    
+    public String getJndiPassword() {
+        return jndiPassword;
+    }
+
+    public void setJndiPassword(String jndiPassword) {
+        this.jndiPassword = jndiPassword;
+    }
+
+    public DestinationType getDestinationType() {
+        return destinationType;
+    }
+
+    public void setDestinationType(DestinationType destinationType) {
+        this.destinationType = destinationType;
+    }
+
+    public String getConnectionFactoryName() {
+        return connectionFactoryName;
+    }
+
+    public void setConnectionFactoryName(String connectionFactoryName) {
+        this.connectionFactoryName = connectionFactoryName;
+    }
+
+    public String getDestinationName() {
+        return destinationName;
+    }
+
+    public void setDestinationName(String destinationName) {
+        this.destinationName = destinationName;
+    }
+
+    public String getDestinationUsername() {
+        return destinationUsername;
+    }
+
+    public void setDestinationUsername(String destinationUsername) {
+        this.destinationUsername = destinationUsername;
+    }
+
+    public String getDestinationPasswordRef() {
+        return destinationPasswordRef;
+    }
+
+    public void setDestinationPasswordRef(String destinationPasswordRef) {
+        this.destinationPasswordRef = destinationPasswordRef;
+    }
+
+    public String getDestinationPassword() {
+        return destinationPassword;
+    }
+
+    public void setDestinationPassword(String destinationPassword) {
+        this.destinationPassword = destinationPassword;
+    }
+
+    @Override
+    public void postLoad(String entityKey, Bundle bundle, @Nullable File rootFolder, IdGenerator idGenerator) {
+        if (getJndiPasswordRef() != null && getJndiPassword() != null) {
```

Comment:
Yes, if the entity contains empty strings for both plain text password and password reference, then I expect it to fail during load. 
They should only be providing one of plain text password or password reference, and not both, regardless if the value is an empty string.
Also, if they provide an empty string for password reference, it will fail during linking.